server.py

```diff
@@ -228,3 +231,42 @@ def semesterIDType(reqstr):
 
   # Convert Officer objects back to dictionaries before returning
   return jsonify([x.__dict__ for x in officers])
+
+#I'm keeping these global variables here for now
+# so it's easier to determine that it goes with the events api
+# when the pieces of the api are inevitably split up
+EV = []
+EVENTS_CACHED = datetime.now()
+def cacheEvents():
+  """
+  Caches the event calendar.
+  """
+  calurl = 'https://calendar.google.com/calendar/ical/ca149os3pmnh0dcopr1jn2negg%40group.calendar.google.com/public/basic.ics'
+  dtstrformat = '%Y-%m-%dT%H:%M:%S.%fZ'
+  global EV, EVENTS_CACHED
+
+  try:
+    cal_request = requests.get(calurl)
+    cal_request.raise_for_status() #requests doesn't raise HTTPError on its own
+    gcal = Calendar.from_ical(cal_request.content)
+    EV = []
+    for comp in [x for x in gcal.walk() if x.name == "VEVENT"]:
```

Comment:
The list comprehension feels redundant to me, but hey, it's idiomatic. We should probably be doing some kind of date-based filtering either here or in the other function.

Reply:
The list format is for ease of management within the backend. I was planning on replacing the list of dicts with a list of objects where the dates can be stored as datetime objects so they can be filtered easily from within the cache.

Reply:
I was talking about the `[x for x in ...` bit, not `EV`.

Reply:
Looking at it now I agree that this isn't the place to do most filtering, and you already do the thing where you filter out past events, so this part looks good.